sentry_streams/sentry_streams/adapters/arroyo/reduce_delegate.py

```diff
@@ -113,17 +114,18 @@ class ReduceDelegateFactory(RustOperatorFactory[TIn, TOut], Generic[TIn, TOut]):
     Creates a `ReduceDelegate`. This is the class to provide to the Rust runtime.
     """
 
-    def __init__(self, step: Reduce[Any, Any, Any]) -> None:
+    def __init__(self, step: Reduce[Any, Any, Any], step_config: ReduceConfig) -> None:
         super().__init__()
         self.__step = step
+        self.__step_config = step_config
 
     def build(self) -> ReduceDelegate[TIn, TOut]:
         retriever = OutputRetriever[TOut]()
         route = Route(source="dummy", waypoints=[])
 
         return ReduceDelegate(
             build_arroyo_windowed_reduce(
-                self.__step.windowing,
+                self.__step.windowing(self.__step_config.batch_size),
```

Comment:
This is the wrong abstraction. 
There are multiple types of window algorithms, not all of them have a batch size. 
It is only the `Batch` dataclass in pipelines for which batch size makes sense.

Ultimately I think the config should not go this deep in the stack. 
You should be able to transform the step object by overriding its configuration inside the adapter (rust_arroyo.py) so that the step object, at that point, already has the right value.